python/ray/data/tests/test_dataset.py

```diff
@@ -482,6 +484,129 @@ def test_tensor_array_validation():
         TensorArray([object(), object()])
 
 
+def test_arrow_ragged_tensor_array_roundtrip():
+    shapes = [(2, 2), (3, 3), (4, 4)]
```

Comment:
can we also have a test that provides multi-dimension ragged ndarray and ensure it throw right exception type + message as expected ?